src/wallet/wallet.cpp

```diff
@@ -226,6 +228,36 @@ CKeyID Wallet::CreateNewKey(bool compressed) {
     return addr;
 }
 
+std::vector<CKeyID> Wallet::GetAllAddresses() {
+    std::vector<CKeyID> result;
+    result.reserve(keyBook.size());
+
+    for (auto it = keyBook.begin(); it != keyBook.end(); it++) {
+        result.push_back(it->first);
+    }
+
+    return result;
+}
+
+std::unordered_map<std::string, std::vector<TxOutput>> Wallet::GetTxoutsWithAddr() {
+    std::unordered_map<std::string, std::vector<TxOutput>> mAddr2Outputs;
+
+    for (const auto& keypair : unspent) {
+        const auto addr  = EncodeAddress(std::get<0>(keypair.second));
+        const auto entry = mAddr2Outputs.find(addr);
+        if (entry != mAddr2Outputs.end()) {
+            VStream vst;
+            vst << addr;
+            Tasm::Listing listing(vst);
+            entry->second.emplace_back(std::get<3>(keypair.second), std::move(listing));
+        } else {
+            mAddr2Outputs.emplace(addr, std::vector<TxOutput>{});
```

Comment:
I think the vector TxOutput should not be empty. It should be added one element from the unspent.